src/config/rollup.config.js

```diff
@@ -88,8 +88,19 @@ if (isPreact) {
 }
 
 const externalPattern = new RegExp(`^(${external.join('|')})($|/)`)
-const externalPredicate =
-  external.length === 0 ? () => false : id => externalPattern.test(id)
+
+function externalPredicate(id) {
+  const isDep = externalPattern.test(id)
+  if (umd) {
+    // for UMD, we want to bundle all non-peer deps
+    return isDep
+  }
+  // for esm/cjs we want to make all node_modules external
+  // TODO: support bundledDependencies if someone needs it ever...
+  const isNodeModule = id.includes('node_modules')
+  const isRelative = id.startsWith('.')
+  return isDep || (!isRelative && !path.isAbsolute(id)) || isNodeModule
```

Comment:
> (!isRelative && !path.isAbsolute(id))

what cases are covered by this which are not covered by `isDep || isNodeModule`?

Reply:
Yeah, it's strange, but this would be called with '@emotion/styled-base' (not listed as a dep because it's added via a babel plugin). So I also need to factor in whether it's relative.

But I can't JUST do relative either because for each mistake it's called with both the source string id and the full path to the resolved module.

Reply:
hm, maybe its brain fart but I still dont get it 😅 non-relative paths can be either:
- package name
- absolute path (which should include node_modules)
- alias (which is not covered, and arguably cant be, here)

am i missing something? 🤔

Reply:
It's kinda hard to explain. All I can say is that if I didn't have the relative and absolute logic in here, then weird things would happen :/

Reply:
it shouldnt hurt, so im not opposed to it, was rather just curious why this is needed

APPROVED! 👍